src/app/index.ts

```diff
@@ -515,6 +515,11 @@ module.exports = class TheiaExtension extends Base {
                     }
                 });
             } else {
+                if ((this.options as any).electron) {
+                    /* run yarn once. This will update the electron version, which requires a second run */
+                    this.spawnCommandSync('yarn', []);
+                }
```

Comment:
@jfaltermeier I don't think this is a viable solution, having to run `yarn` twice for `electron` is not ideal, and when users generate their extensions they get build errors which will only add to the confusion:

- https://github.com/eclipse-theia/generator-theia-extension/actions/runs/4872257762/jobs/8690217178?pr=167#step:5:110

Reply:
Ok, but if we don't run yarn twice as part of the generator and the electron version available in the template does not match the electron version Theia wants (either because a newer Theia version updates electron or the user specified a different/older theia-version for generation), then the generation will just fail with the same error: https://github.com/eclipse-theia/generator-theia-extension/actions/runs/4866599386/jobs/8678299442#step:5:110 

So this is why I preferred this approach. 

Would it be better if we
a) add the electron version that is currently required to the template,
b) only run yarn twice in the electron case if there was an error, and
c) add some log messages explaining the second run?

Reply:
I'd be fine with keeping the behavior as is and making sure that the `electron` version matches the `peerDependencies` version specified by `@theia/electron`. It's the same thing we do for typescript, and node where we match the version required by the framework.

If not I think a solution that does not need a full `yarn` after the fact is best (ex: we get the error, we get duplicated logs for browser-app) as it does not give a good experience. For the most part we have been following the idea that users that generate extensions will use the latest version, not some old version else you might encounter problems.

Reply:
:+1: Thanks, I've updated the PR